server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -1399,19 +1386,10 @@ private static void validateCompositeTemplate(
 
     private static void validateTemplate(
         Settings validateSettings,
-        String mappings,
+        CompressedXContent mappings,
         IndicesService indicesService,
         NamedXContentRegistry xContentRegistry
     ) throws Exception {
-        // First check to see if mappings are valid XContent
-        if (mappings != null) {
-            try {
-                new CompressedXContent(mappings);
-            } catch (Exception e) {
-                throw new MapperParsingException("Failed to parse mapping: {}", e, mappings);
```

Comment:
Actually I think it's outright impossible to ever run into this issue (even up the stack). We only ever get this string out of `org.elasticsearch.action.admin.indices.template.put.PutIndexTemplateRequest#mappings`. But there it is validated by parsing and reformatting anyway (even more than by just compressing). So I think this already is dead code and would never have reached the end-user because setting up `PutIndexTemplateRequest` would have failed before.

Reply:
> I think it's outright impossible to ever run into this issue (even up the stack)

I see now, thanks for the clarification.